src/Microsoft.TestPlatform.CoreUtilities/Helpers/EnvironmentHelper.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CoreUtilities.Helpers
+{
+    using System;
+    using ObjectModel;
+
+    public class EnvironmentHelper
+    {
+        public const string VstestConnectionTimeout = "VSTEST_CONNECTION_TIMEOUT";
+        public const int DefaultConnectionTimeout = 90; // seconds
+
+        /// <summary>
+        /// Get timeout based on environment variable VSTEST_CONNECTION_TIMEOUT.
+        /// </summary>
+        public static int GetConnectionTimeout()
+        {
+            var envVarValue = Environment.GetEnvironmentVariable(EnvironmentHelper.VstestConnectionTimeout);
+            if (!string.IsNullOrEmpty(envVarValue) && int.TryParse(envVarValue, out int value))
```

Comment:
if user sets a very small value say 10, should we honor it? I would say we keep 90 or 60 something as min, & users value will only be honored if it is > min val
More over here we are not even checking for -ve values

Reply:
Handled -ve case. Keep min value 1, giving more choice to user.